core/rfb.js

```diff
@@ -548,19 +542,7 @@
                     Util.Error("Got data while disconnected");
                     break;
                 case 'normal':
-                    if (this._normal_msg() && this._sock.rQlen() > 0) {
-                        // true means we can continue processing
-                        // Give other events a chance to run
-                        if (this._msgTimer === null) {
-                            Util.Debug("More data to process, creating timer");
-                            this._msgTimer = setTimeout(function () {
-                                this._msgTimer = null;
-                                this._handle_message();
-                            }.bind(this), 0);
-                        } else {
-                            Util.Debug("More data to process, existing timer");
-                        }
-                    }
+                    while (this._normal_msg() && this._sock.rQlen() > 0);
```

Comment:
Hmm... I like this in principle, but I'm worried about getting a large number of messages and then causing the browser to block.  I'd like @kanaka's thoughts on this one, but I think I'm tentatively ok with it.